data-products/src/recommendation_api/corpus_candidate_sets_flow.py

```diff
@@ -93,12 +93,23 @@ async def load_corpus_candidate_set_records(
     )
     sql_query = Path(sql_path).read_text()
 
-    corpus_items = await snowflake_query(
-        query=sql_query,
-        snowflake_connector=snowflake_connector,
-        params=candidate_set_config.query_params,  # type: ignore
-        cursor_type=DictCursor,  # type: ignore
-    )
+    if candidate_set_config.is_multiquery:
+        # if multiquery we need to use different function
+        sql_split = sql_query.split(";")
+        results = await snowflake_multiquery(
+            queries=sql_split[:-1],  # omit last item which will be blank
+            snowflake_connector=snowflake_connector,
+            params=candidate_set_config.query_params,  # type: ignore
+            cursor_type=DictCursor,  # type: ignore
+        )
+        corpus_items = results[-1::][0]  # only need the last set of results
```

Comment:
I appreciate how you often go for the clean, long-term solution.

If [recommended_by_recency.sql](https://github.com/Pocket/data-flows/pull/292/files#diff-188289ca819d7b78d103190dbaff850db48f73166a7920d8a8f6dc2dd45be550) is the only case where we need multiqueries, I would've simply refactored that SQL into a single query. For example by duplicating `max_timestamp` in-place, or by adding a `with max_timestamp as ( ... )` to join against.

Reply:
Thank You